compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -361,6 +366,17 @@ fn collect_items_rec<'tcx>(
     recursion_limit: Limit,
     mode: CollectionMode,
 ) {
+    if state.error_count.load(Ordering::Relaxed) > MAX_NUMBER_OF_ERRORS {
+        tcx.dcx().span_err(
+            starting_item.span,
+            format!(
+                "the limit of {} errors has been reached during monomorphization",
+                MAX_NUMBER_OF_ERRORS
+            ),
+        );
```

Comment:
This, if approved, better become something like:
```rust
#[derive(Diagnostic)]
#[diag(monomorphize_encountered_too_errors)]
pub(crate) struct EncounteredTooError {
    #[primary_span]
    pub span: Span,
    pub formatted_item: String,
    /// ...
}
```
in `rustc_monomorphize/src/errors.rs`.